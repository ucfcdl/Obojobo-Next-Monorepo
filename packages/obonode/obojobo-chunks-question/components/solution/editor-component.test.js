import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import Solution from './editor-component'

import { Transforms } from 'slate'
jest.mock('slate')
jest.mock('slate-react')
jest.mock(
	'obojobo-document-engine/src/scripts/oboeditor/components/node/with-slate-wrapper',
	() => item => item
)

describe('Solution Editor Node', () => {
	test('Solution component', () => {
		const component = renderer.create(<Solution />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('Solution component deletes self', () => {
		const component = mount(<Solution element={{ content: {} }} />)

		component.find('button').simulate('click')

		expect(Transforms.removeNodes).toHaveBeenCalled()
	})
})
