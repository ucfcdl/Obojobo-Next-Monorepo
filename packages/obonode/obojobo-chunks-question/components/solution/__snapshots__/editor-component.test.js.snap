// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Solution Editor Node Solution component 1`] = `
<div
  className="solution-editor"
>
  <div
    className="obojobo-draft--components--button is-not-dangerous align-center delete-button"
    contentEditable={false}
  >
    <button
      className="button"
      contentEditable={false}
      onClick={[Function]}
    >
      Ã—
    </button>
  </div>
</div>
`;
